include/pmacc/particles/memory/boxes/ParticlesBox.hpp

```diff
@@ -330,7 +330,7 @@ class ParticlesBox : protected DataBox<PitchedBox<SuperCell<T_Frame>, DIM> >
         return false;
     }
 
-    HDINLINE SuperCellType& getSuperCell( DataSpace<DIM> idx )
+    HDINLINE SuperCellType& getSuperCell( DataSpace<DIM> idx ) const
```

Comment:
This method is technically constant, but seems to be logically not constant, as it returns a reference to some internal stuff. In my mind, maybe a better solution is to leave the old method as is and add a const twin, which returns a const reference and then use that new one?

I am also not sure about some others changed, maybe the similar argument applies.

Reply:
If it compiles it's valid C++ and `const` on a method technically only means the method causes no internal side effects.
I think it's ok for now to progress on what he is implementing.